transformation/src/main/scala/org/broadinstitute/monster/hca/HcaLog.scala

```diff
@@ -0,0 +1,60 @@
+package org.broadinstitute.monster.hca
+
+import org.broadinstitute.monster.hca.PostProcess.errorCount
+import org.slf4j.Logger
+import ujson.Obj
+
+// a simple log level case class hierarchy to make sure we log at the right level
+trait LogLevel
+case class HcaErrorLog() extends LogLevel
+case class HcaWarnLog() extends LogLevel
+
+// a trait to capture the generic logging mechanism we'll want, with case classes for the different logging levels
+abstract class HcaLog {
+  val level: LogLevel
+  val jsonMsg: Obj
+
+  def log(implicit logger: Logger): Unit =
+    level match {
+      case HcaErrorLog() =>
+        logger.error(jsonMsg.toString())
+        errorCount.inc()
+      case HcaWarnLog() => logger.warn(jsonMsg.toString())
+    }
+}
+
+class HcaWarn(msg: String) extends HcaLog {
+  val level: HcaWarnLog = HcaWarnLog()
+
+  val jsonMsg: Obj = ujson.Obj(
+    "warningType" -> ujson.Str(this.getClass.getSimpleName),
+    "message" -> ujson.Str(msg)
+  )
+}
+
+class HcaError(filepath: String, msg: String) extends HcaLog {
+  val level: HcaErrorLog = HcaErrorLog()
+
+  // match everything that is not followed by a "/" (only the filename)
+  private val filenameRegex = "[^/]*$".r
+  private val filename = filenameRegex.findFirstIn(filepath).getOrElse("")
+
+  val jsonMsg: Obj = ujson
+    .Obj(
+      "errorType" -> ujson.Str(this.getClass.getSimpleName),
+      "filePath" -> ujson.Str(filepath),
+      "fileName" -> ujson.Str(filename),
+      "message" -> ujson.Str(msg)
+    )
+}
+
+// case classes for all the different actual warnings and errors we want to raise during the workflow
+case class NoMatchWarning(msg: String) extends HcaWarn(msg)
```

Comment:
Have we hit instances of each of these types of warnings/errors?

Reply:
I added warnings and errors for things we've been encountering and things we agreed to in the MVP spec.